### PR TITLE
Fixes for build failures when using MinGW

### PR DESCRIPTION
**Pull request by walbourn**
This fixes build failures with the latest release when using the MinGW toolset.

### CODE REVIEW

**Comment by Biswa96** [comment]
Would you like to share any compiler or linker output related to that build failure? Just curious.

**Comment by walbourn** [comment]
```
c:\mingw64\x86_64-w64-mingw32\include\rpcndr.h:15:2: error: #error incorrect <rpcndr.h> version. Use the header that matches with the MIDL compiler.
   15 | #error incorrect <rpcndr.h> version. Use the header that matches with the MIDL compiler.
```

**Comment by walbourn** [comment]
Working on fixing up the GoogleTest as well for MinGW.

**Comment by Biswa96** [comment]
Thanks for the info. I thought that version mismatch was fixed in https://github.com/microsoft/DirectX-Headers/pull/93

File: src/d3dx12_property_format_table.cpp [hunk 1]
```diff
@@ -12,7 +12,10 @@
     #define WIN32_LEAN_AND_MEAN
 #endif
 #ifndef NOMINMAX
-    #define NOMINMAX
+    #define NOMINMAX 1
+#endif
+#ifdef __MINGW32__
```
Comment (jenatali) [inline src/d3dx12_property_format_table.cpp:17]: This doesn't need to be `#ifdef`'d, it's always safe to include. You can just put it after the `winadapter.h` include.

Comment (walbourn) [reply src/d3dx12_property_format_table.cpp:17]: It's mostly to make it clear that it's a workaround for MinGW.

**Pull request merged**